ion_schema_2_0/constraints/ieee754_float.isl

```diff
@@ -0,0 +1,378 @@
+$ion_schema_2_0
+
+type::{
+  name: ieee754_float_binary16,
+  ieee754_float: binary16,
+}
+
+$test::{
+  type: ieee754_float_binary16,
+  should_accept_as_valid: [
+    0e0,
+    -0e0,
+    1e0,
+    -1e0,
+    nan,
+    +inf,
+    -inf,
+
+    /////////////////////////////////////////////////
+    //////  Special numbers for 16-bit floats  //////
+    /////////////////////////////////////////////////
+
+    // Smallest positive subnormal
+    5.960464477539063e-8,
+    -5.960464477539063e-8,
+    // Smallest positive normal
+    6.103515625e-5,
+    -6.103515625e-5,
+    // Closest to +/- 1/3
+    0.333251953125e0,
+    -0.333251953125e0,
+    // Next numbers above and below +/-1
+    0.99951171875e0,
+    1.0009765625e0,
+    -0.99951171875e0,
+    -1.0009765625e0,
+
+    2047e0,
+    2048e0, // <-- Largest contiguous integer
+    2050e0,
+
+    // Largest magnitude representable numbers
+    65504e0,
+    -65504e0,
+  ],
+  should_reject_as_invalid: [
+    null.float,
+    1,
+    2d0,
+    [],
+    {},
+    (1e0),
+    "Hello",
+    'World!',
+    2022-03-04T,
+
+    /////////////////////////////////////////////////
+    //////  Special numbers for 16-bit floats  //////
+    /////////////////////////////////////////////////
+
+    2049e0, // <-- First non-mathcing integer
```

Comment:
non-matching?

Reply:
You've never heard of "mathcing" before? _cing_ is Albanian for "to touch", so what I'm really saying is that this is the first integer that is not math-touching the 16-bit floating point number format.